crates/sqlexec/src/parser/options.rs

```diff
@@ -202,6 +203,18 @@ impl fmt::Display for StmtOptions {
     }
 }
 
+impl TryFrom<&mut StmtOptions> for StorageOptions {
+    type Error = ParserError;
+
+    fn try_from(value: &mut StmtOptions) -> Result<Self, Self::Error> {
+        let mut inner = BTreeMap::new();
+        for (key, value) in value.m.iter() {
+            inner.insert(key.clone(), value.clone().parse_opt()?);
+        }
```

Comment:
We could also perform validation here that the key is one of:
- [Azure options](https://docs.rs/object_store/latest/object_store/azure/enum.AzureConfigKey.html#variants)
- [S3 options](https://docs.rs/object_store/latest/object_store/aws/enum.AmazonS3ConfigKey.html#variants)
- [Google options](https://docs.rs/object_store/latest/object_store/gcp/enum.GoogleConfigKey.html#variants)

Reply:
It would be a programmer error if we had another type though, correct?

Reply:
Not necessarily; it would for example allow us to propagate some other props that could be internally useful for us, but not for building the object store itself (also the key is actually just a `String` here, I'm just considering whether to validate the possible values of that string).